src/Database/Type/IntegerType.php

```diff
@@ -36,7 +36,7 @@ class IntegerType extends BaseType implements BatchCastingInterface
      */
     protected function checkNumeric(mixed $value): void
     {
-        if (!is_numeric($value)) {
+        if (!is_numeric($value) && !is_bool($value)) {
```

Comment:
FloatType does not perform this check. These logic for these types should probably be aligned.